### DISCUSSION TITLE
Adding surface-based Schaefer atlas

### DISCUSSION THREAD

**Discussion by kazinka** [simnibs/simnibs#255] [9. Results and analysis] [answered]
I am trying to examine the electric field potential for my head models using standardized parcellations so that I can make direct comparisons to fMRI. I have been able to accomplish this using the Glasser parcellations. However, I would like to use the Schaefer atlas instead, as that is what I'm using for my fMRI analyses. As far as I can tell, this atlas is not already available in simNIBS (currently has Destrieux atlas, Desikan-Killiany atlas, and HCP/Glasser atlas). Is there a way to add other atlases? I have a dlabel.nii file that I used for the fMRI analyses.

Below is my matlab code to get the Glasser parcellations, based on [sample code from simNIBS](https://simnibs.github.io/simnibs/build/html/tutorial/advanced/group_analysis.html): 

```
[m, snames]=mesh_load_fssurf('fsaverage','label','HCP_MMP1');
region_names=snames;%create` a setup so that we get all the ROIs
for r = [2:181 183:length(region_names)] %ignore blank region names
    region_name = region_names{r};
    roi_idx=find(strcmpi(snames, region_name));
    node_idx = m3.node_data{1,1}.data==roi_idx;
    % visualize region
    %m3.node_data{end+1}.data = int8(node_idx);
    %m3.node_data{end}.name = region_name;
    %mesh_show_surface(m3, 'field_idx', region_name)

    % calculate a weighted mean using the node areas
    nodes_areas = mesh_get_node_areas(n);
    avg_field_roi = sum(avg_field(node_idx).*nodes_areas(node_idx))/sum(nodes_areas(node_idx));
    fprintf('Average %s in %s: %f\n', field_name, region_name, avg_field_roi)

    %calculate mean for each person
    for i = 1:size(fields_n,2)
        %if ~isempty(subs{i})
            %Ind_field_roi(i) = sum(fields(node_idx,i).*nodes_areas(node_idx))/sum(nodes_areas(node_idx));
            Ind_field_roi_n(r,i) = sum(fields_n(node_idx,i).*nodes_areas(node_idx))/sum(nodes_areas(node_idx));
        %else 
        %    Ind_field_roi(r,i) = NaN;
        %end
    end
end
```

**Comment by jduemose** [comment, marked as answer]
Hi

You will need to have the atlas available in the same format as the other atlases (freesurfer's "annot" files). I don't know if that is available or if you will need to create that yourself (e.g., sampling the values in the nifti file at the vertices)? 

Once you have that, it should be pretty straightforward: you put the atlas file in the same folder as the other atlases (simnibs/resources/templates/fsaverage_atlases) and add another `elseif` clause to [this](https://github.com/simnibs/simnibs/blob/e15afab5712838fe728e62f29505172f8d096fd8/simnibs/matlab_tools/mesh_load_fssurf.m#L144) `if` block with the name of your new atlas.

That way you should be able to use `mesh_load_fssurf`.

Best,
Jesper
